src/dsp/rx_meter.cpp

```diff
@@ -55,11 +56,12 @@ int rx_meter_c::work(int noutput_items,
 {
     std::lock_guard<std::mutex> lock(d_mutex);
 
-    const gr_complex *in = (const gr_complex *) input_items[0];
-    (void) output_items; // unused
-
-    for (int i = 0; i < noutput_items; i++)
-        d_cbuf.push_back(in[i]);
+    if (noutput_items > d_writer->bufsize())
+        noutput_items = d_writer->bufsize();
```

Comment:
I don't think this is right. This block is meant to skip over samples that cannot fit in the circular buffer. Reducing `noutput_items` will tell the GNU Radio scheduler that they haven't been consumed.

Reply:
I have not removed it after attempting to reduce memory usage by shrinking the buffer to 2 * d_fftsize. This was done to avoid writing behind the end of the buffer.
If we tell GNU Radio scheduler that some items are not consumed, then GNU Radio scheduler will call `work` again until all items get consumed, so no problem here.
Shrinking the buffer did not work as expected - artifacts at low FFT size and low input rate, so I reverted the change, but forgot to remove this safety measure.
I still think, that we can safely limit the buffer to 2 * max FFT size and not waste ~160Mb and more of shared memory in case of HackRF/ LimeSDR at high sampling rate. But this should go into separate PR.

Reply:
> If we tell GNU Radio scheduler that some items are not consumed, then GNU Radio scheduler will call `work` again until all items get consumed, so no problem here.

True, but there will be extra overhead, won't there? We could instead just skip over the excess input samples.